### ISSUE TITLE
Scrolling behavior of the "Contents" sidebar is inaccurate, doesn't detect current section

### ISSUE THREAD

**Issue by grisaitis** [executablebooks/sphinx-book-theme#607]
### Describe the bug

**context**
When I click on a section name under the "Contents" sidebar on the right...

**expectation**
I expected that section to be highlighted and any subsections expanded under the name.

**bug**
But instead:
- the previous section shows up as currently viewed.
- the problem is "solved" if i scroll down a little 
   - (ie, only after scrolling down a smidge does the selected section become highlighted and expanded under "Contents")

**problem**
This is a problem for people navigating pages via the table of contents sidebar. Using the sidebar is useful for browsing large pages, but this issue makes navigation less user-friendly and sometimes very confusing.

### Reproduce the bug

this can be seen on any site built with this sphinx theme. For example, in the docs for this project, it can be seen at: https://sphinx-book-theme.readthedocs.io/en/stable/customize/sidebar-primary.html#customize-the-sidebar-footer

**addendum - potential causes**
- when selecting a section (e.g. opening a URL like `page.html#section-name`), i think the page needs to be scrolled down a few pixels more. Currently, the last few pixels of the previous section are still visible.
- Maybe this is related to #448 and the [IntersectionObserver](https://developer.mozilla.org/en-US/docs/Web/API/IntersectionObserver) API. Maybe a pixel offset needs to be adjusted somewhere?
   - I tried modifying `sphinx-book-theme.js` to have `e[0].boundingClientRect.y<-20`, or also `-200`, but noticed no change in the ui.

**note**
i opened an issue about this at https://github.com/executablebooks/jupyter-book/issues/1827. But I think this repo is the correct place to open this.

**Comment by choldgraf** [comment]
Yep - thanks for noticing this. I believe this is a duplicate of the issue below, feel free to re-open if you think it's different!

- https://github.com/executablebooks/sphinx-book-theme/issues/545

**Comment by grisaitis** [comment]
thank you! yes, it is indeed a dupe.
